bap-vibes/src/core_c.ml

```diff
@@ -143,262 +139,240 @@ module Eval(CT : Theory.Core) = struct
       hvars;
     }
 
-  let char_ty = Theory.Bitv.define 8
-  let short_ty = Theory.Bitv.define 16
-  let long_ty = Theory.Bitv.define 32
-  let long_long_ty = Theory.Bitv.define 64
+  let char_ty = T.Bitv.define 8
+  let short_ty = T.Bitv.define 16
+  let long_ty = T.Bitv.define 32
+  let long_long_ty = T.Bitv.define 64
```

Comment:
Is there a way to obtain this information based on the target, or do we have to add this manually?

Reply:
There's definitely no compiler-provenance analysis, so you would need to specify the compiler manually. There's the `Theory.Unit.compiler` in the unit class that you can use, but its purpose was to capture the configuration of the tool that was used to generate the unit's code rather than to describe the compiler model that is necessary for the code generation tool to imitate the behavior of a specific compiler. I think such information is out of the scope of the binary analysis tool and is more in the scope of the code-generation tool. In other words, I would suggest you add the compiler model into vibes (add it to the state of your elaboration state monad) and specify it manually through your configuration. It is also worthwhile to look into bap-c in order to reuse its functionality. In addition, we can think about leveraging the ABI infrastructures as it captures much of the needed information. For example, you can look at the `Bap_abi.name` attribute of the project to get the ABI-name, which is already something.